src/Microsoft.Graph.Core/Requests/BaseRequest.cs

```diff
@@ -456,5 +455,17 @@ private string GetHeaderValue(HttpRequestMessage requestMessage, string headerNa
 
             return headerValue;
         }
+
+        /// <summary>
+        /// Checks if a custom IHttpProvider is used or our HttpProvider is used without an auth handler (flag).
+        /// </summary>
+        /// <returns></returns>
+        private bool ShouldAuthenticateRequest()
+        {
+            if (this.Client.HttpProvider.GetType() != typeof(HttpProvider))
```

Comment:
It's ok the way it is, since we are checking if a customer is using our default `HttpProvider` with an `AuthHandler` set (we let the auth handler authenticate) or if they using a custom or default HttpProvider without an AuthHandler (we let `BaseRequest` authenticate). Let me re-write it to something like this to make it clear:
```cs
 if (this.Client.HttpProvider.GetType() == typeof(HttpProvider) && (this.Client.HttpProvider as HttpProvider).httpClient.ContainsFeatureFlag(FeatureFlag.AuthHandler))
    return false;
else
    return true;
```